US26 - Page detail for tags, fix #126

davidbgk opened the pull request:
ETQ lecteur, je veux consulter les articles pour un tag donné

- [x] - doit permettre de visualiser la liste des articles pour un tag donné (par ordre antéchrono) : photo de couverture, titres
- [x] - ne doit renvoyer que les articles dans la langue de l'interface
- [x] - peut afficher la photo et/ou la description du tag
- [x] - doit comporter une url partageable de type _/tag/meetmyhood_

davidbgk committed:
US26 - Page detail for tags, fix #126

davidbgk committed:
Merge branch '70-tags-addition' into 126-tags-browsability

davidbgk committed:
Merge branch '70-tags-addition' into 126-tags-browsability

davidbgk committed:
Merge branch '70-tags-addition' into 126-tags-browsability

davidbgk committed:
Merge branch '70-tags-addition' into 126-tags-browsability

davidbgk committed:
US22 - Tags administration, fix #30

davidbgk committed:
Merge branch '70-tags-addition' into 126-tags-browsability

davidbgk committed:
Tag has a name and not a title, copy-pasta failure

davidbgk committed:
Merge branch '126-tags-browsability' into 30-tags-administration

davidbgk committed:
Merge branch '70-tags-addition' into 126-tags-browsability

davidbgk committed:
Merge branch '126-tags-browsability' into 30-tags-administration

vinyll commented on cafebabel/templates/articles/tags/detail.html:
```diff
@@ -0,0 +1,20 @@
+{% extends "base.html" %}
+
+{% block meta_title %}{{ tag.name }}{% endblock %}
+{% block meta_description %}{{ tag.summary }}{% endblock %}
```
These probably need to be enriched with the extra metas (ie: the image)

davidbgk replied:
@cafebabel-webmaster can you take care of this?

johangiraud replied:
ok done.

vinyll commented on cafebabel/tests/test_articles.py:
```diff
@@ -359,4 +359,5 @@ def test_article_detail_contains_tags(client, app, tag, published_article):
     assert response.status_code == HTTPStatus.OK
-    assert 'Wonderful, Sensational.' in response
+    assert '<a href=/article/tag/wonderful/>Wonderful</a>,' in response
+    assert '<a href=/article/tag/sensational/>Sensational</a>.' in response
```
Wondering if the comma end the final point should be written in the test as the HTML may change into a list or a different display.

davidbgk replied:
I kept it because it tests conditioning within templates. It's OK to me to change tests when you change the frontend.

vinyll replied:
I see it like the punctuations are not a condition for the test to be valid or not and it may be disturbing for the frontend to have the test broken while not feeling like breaking something.
No big deal anyways, let's keep it then.

vinyll commented on cafebabel/tests/test_tags.py:
```diff
@@ -74,3 +74,24 @@ def test_tag_suggest_wrong_language(client, tag):
     assert response.status_code == HTTPStatus.BAD_REQUEST
     assert (b"Languages available: ['en', 'fr', 'es', 'it', 'de']"
```
We could update this to compare with `… in response`

davidbgk replied:
I don't get it, what is your suggestion? can you elaborate? (by the way, that's not the PR related to that test)

davidbgk replied:
Ah! Never mind, I think I get it, I'll change that upstream.

davidbgk replied:
Done in 78dd9e382303cec364fa1658bcb82dbb3343ddad

vinyll committed:
tag editing now takes the current language

johangiraud committed:
styles detail tag page + parallax on picture

johangiraud committed:
clean up test css values

johangiraud committed:
wip styles edit tag page

johangiraud committed:
cleaning after review + change to more precise js file name

johangiraud committed:
styles article list

johangiraud committed:
add specific metas for tag page

johangiraud committed:
RE styles detail tag page + parallax on picture

johangiraud committed:
RE wip styles edit tag page

johangiraud committed:
RE cleaning after review + change to more precise js file name

johangiraud committed:
RE styles article list

johangiraud committed:
RE add specific metas for tag page

johangiraud committed:
comment edit tag page button

johangiraud committed:
REMOVE add specific metas for tag page

johangiraud committed:
REMOVE styles article list

johangiraud committed:
REMOVE cleaning after review + change to more precise js file name

johangiraud committed:
REMOVE wip styles edit tag page

johangiraud committed:
REMOVE styles detail tag page + parallax on picture

johangiraud committed:
update js file name

johangiraud committed:
correction js file name in template

johangiraud committed:
removing meta block and edit button

johangiraud committed:
put back edition button

davidbgk committed:
Remove link to edition (not par of that branch)

davidbgk committed:
Merge branch '70-tags-addition' into 126-tags-browsability

# Conflicts:
#	cafebabel/tests/test_tags.py

davidbgk committed:
Merge branch '126-tags-browsability' into 30-tags-administration

# Conflicts:
#	cafebabel/templates/articles/tags/detail.html

davidbgk committed:
Prevent the edition of tag name and language

Indirectly the slug becomes permanent too which solves a couple of nasty side-cases (redirections and so on)

davidbgk committed:
Merge branch '70-tags-addition' into 126-tags-browsability

davidbgk committed:
Merge branch '126-tags-browsability' into 30-tags-administration

vinyll committed:
Tag detail filters language

davidbgk committed:
Merge branch '70-tags-addition' into 126-tags-browsability

davidbgk committed:
Ensure fallback to a default language

davidbgk committed:
Only link to tags if article is published

davidbgk committed:
Link in article cards only on the title, fix #155

davidbgk committed:
Merge branch '126-tags-browsability' into 30-tags-administration

# Conflicts:
#	cafebabel/articles/tags/views.py

johangiraud committed:
set parallax effect to half of image size

johangiraud committed:
add article link on thumbnail

johangiraud committed:
Merge branch '126-tags-browsability' into 30-tags-administration

johangiraud committed:
move edit button below tag title

johangiraud committed:
Merge branch '30-tags-administration' of github.com:cafebabel/cafebabel.com into 30-tags-administration

johangiraud committed:
Merge pull request #153 from cafebabel/30-tags-administration

US22 - Tags administration, fix #30

johangiraud committed:
Merge branch '70-tags-addition' into 126-tags-browsability

vinyll committed:
tag articles are listed sorted by publication date

vinyll committed:
tag summary is optional

vinyll committed:
restored link on tag for articles and tag display for drafts

davidbgk committed:
Merge branch '70-tags-addition' into 126-tags-browsability

johangiraud committed:
style tag page edit + tag display in article detail

johangiraud committed:
minor detail article corrections

johangiraud committed:
remove useless console.log
